### ISSUE TITLE
Split up the generated OpenApi file into multiple files

### ISSUE THREAD

**Issue by klacol** [RSuter/NSwag#1905]
We generate our OpenApi files (2.0 and 3.0) from our ASP.NET Controllers, as described [here](https://github.com/RSuter/NSwag/wiki/WebApiToSwaggerGenerator). That works like a charm.

Since our models are quite big, this takes a while (about 1 second for generating the Json file and about 7 seconds to render it in the Swagger UI).

We have tested with [ReDoc](https://github.com/ReDoc-ly/create-openapi-repo#generator-openapi-repo) directly. There, the API-specification file is splitted up into several smaller files. They live in subfolder to the specification files (structured in definitions and paths). This results in a more modular structure, and the loading time is shorter.

You can see that in this repo: https://github.com/Rebilly/RebillyAPI

Does NSwag have the possiblity to generate the OpenAPI-file splittet up in supfolders and referenced files?

**Comment by RSuter** [comment]
No this feature is not yet supported. Basic file $refs work but when serialized back you’ll end up with an unified document. 

This feature would need quite some work in order to have a good file abstraction and correct reference resolution when deserializing...

**Comment by klacol** [comment]
Thanks for this clarification. 

In our case, this would only be one-way (Serialize C# to Json) , only to be able to have the HTML docs. A deserialization would not be needed.

**Comment by RSuter** [comment]
Sorry, i meant serialization (updated). How would this look like? A file per dto schema and also per operation?

**Comment by maikebing** [comment]
So, are there any options available or have this plan in the future?

**Comment by RSuter** [comment]
Indeed, you could generate multiple files from one SwaggerDocument, but you’d need to replace the refs into external refs, etc. 

Will work on issues with higher prio first...

**Comment by syndicatedshannon** [comment]
see #1398
